### PR TITLE
Better specs

### PR DESCRIPTION
**Pull request by ntdesmond**
argh.

### CODE REVIEW

File: Docs/Specification/3_Types.md [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+
+This type is a base for all other types. It defines a single method `ToString() : String`, which returns an empty string.
+
+All the built-in classes inherit from this class, or from its descendants.
+
+All the classes in the program in O Language inherit `Class` directly, unless this behavior is changed with the `extends` syntax.
+
+#### `Void`
+
+This type has no logic at all, and represents an absence of the return value of a function.
```
Comment (DOCtorActoAntohich) [inline Docs/Specification/3_Types.md]: `no logic` -> `no behavior`

File: Docs/Specification/5_Generics.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+The O language also features a support of [generic](https://en.wikipedia.org/wiki/Generic_programming) classes, which allow to specify the types of the objects used inside these classes, during the instantiation process, i.e. when the object of the generic class is created.
+
+## Generic class definition
+
+To create a generic class, just list the names of generic types to be used inside the class field definitions and methods in the angle brackets after the class name.
+
+## Using the generic types
+
+To use the defined generic type, just pass it where needed, the same way as usual types (`String`, `Real`, etc.) would be passed.
+
```
Comment (DOCtorActoAntohich) [inline Docs/Specification/5_Generics.md:12]: Add info about type inference rules (simple substitution)

File: Docs/Specification/3_Types.md (hunk 1 above)
Comment (ntdesmond) [reply Docs/Specification/3_Types.md]: `MonoBehaviour` moment

**Comment by ntdesmond** [comment]
fixed both problems

**Comment by ntdesmond** [comment]
oh nice fuck it

UPD: it didn't want to merge branches, I resolved the conflicts with the commit below

**Review by sevenzing — DISMISSED** [review]

**Review by DOCtorActoAntohich — APPROVED** [review]
:trollface:

**Pull request merged**